build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/biomeTasks.ts

```diff
@@ -39,41 +37,26 @@ export class BiomeTask extends LeafWithFileStatDoneFileTask {
 	}
 
 	/**
-	 * Includes all files in the task's package directory and any biome config files in the directory tree. Files ignored
-	 * by git are excluded.
+	 * Includes all files in the the task's package directory that Biome would format and any Biome config files that
+	 * apply to the directory. Files ignored by git are excluded.
 	 */
 	protected async getInputFiles(): Promise<string[]> {
-		const repoRoot = await this.repoRoot;
 		const gitRepo = await this.gitRepo;
+		// Absolute paths to files that would be formatted by biome.
+		const files = await getBiomeFormattedFiles(this.node.pkg.directory, gitRepo);
+		const configPath = await getClosestBiomeConfigPath(this.node.pkg.directory);
 
-		const configFiles = await this.getBiomeConfigPaths(this.node.pkg.directory);
-		const files = (await gitRepo.getFiles(this.node.pkg.directory)).map((file) =>
-			path.join(repoRoot, file),
-		);
+		if (configPath === undefined) {
+			// No configs to include, so just return all formatted files
+			return [...new Set(files)];
+		}
 
-		return [...new Set([...configFiles, ...files])];
+		const allConfigPaths = await getAllBiomeConfigPaths(configPath);
+		return [...new Set([...allConfigPaths, ...files])];
 	}
 
 	protected async getOutputFiles(): Promise<string[]> {
 		// Input and output files are the same.
 		return this.getInputFiles();
```

Comment:
Output files are just the formatted files, right?

Reply:
Correct.

Reply:
Ahhh, I didn't consider that. Great point.